### ISSUE TITLE
i32::from_str_radix doesn't work for negative binary numbers

### ISSUE THREAD

**Issue by sanyamc** [rust-lang/rust#74914]
I tried this code:

```rust
let f = format!("{:b}", -3);
let s = i32::from_str_radix(&f, 2).unwrap();
```
I expected to see this happen: *No Compiler error and s should be -3.*

Instead, this happened: *Got compiler error ParseIntError { kind: Overflow }*

### Meta

Yes; it reproes in nightly too: 1.47-nightly

`rustc --version --verbose`:
```
rustc 1.44.0 (49cae5576 2020-06-01)
binary: rustc
commit-hash: 49cae55760da0a43428eba73abcb659bb70cf2e4
commit-date: 2020-06-01
host: x86_64-unknown-linux-gnu
release: 1.44.0
LLVM version: 9.0
```

<details><summary>Backtrace</summary>
<p>

```
    Finished dev [unoptimized + debuginfo] target(s) in 0.82s
     Running `target/debug/playground`
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseIntError { kind: Overflow }', src/main.rs:3:13
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1076
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1537
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:486
  11: rust_begin_unwind
             at src/libstd/panicking.rs:388
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:101
  13: core::option::expect_none_failed
             at src/libcore/option.rs:1272
  14: core::result::Result<T,E>::unwrap
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libcore/result.rs:1005
  15: playground::main
             at src/main.rs:3
  16: std::rt::lang_start::{{closure}}
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/rt.rs:67
  17: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  18: std::panicking::try::do_call
             at src/libstd/panicking.rs:297
  19: std::panicking::try
             at src/libstd/panicking.rs:274
  20: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  21: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  22: std::rt::lang_start
             at /rustc/5c1f21c3b82297671ad3ae1e8c942d2ca92e84f2/src/libstd/rt.rs:67
  23: main
  24: __libc_start_main
  25: _start
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

</p>
</details>

**Comment by jonas-schievink** [comment]
This is trying to parse `11111111111111111111111111111101`, which is out of range of `i32`, so this behaves as expected. See the docs of `from_str_radix`:

> The string is expected to be an optional + or - sign followed by digits.

**Comment by sanyamc** [comment]
Thanks for quick response! Thats very intersting though...usually signs (-, +) are encoded in the bits itself for binary rather than having to supply like '-11111100000'. For that reason, even though i did read the docs, couldn't fathom that i have to supply the sign for binary representation too. Interesting design choice for i feel because i have to read the first bit to figure out the sign of number then pass it along to the API.
